pkg/core/server/server.go

```diff
@@ -29,24 +32,40 @@ func NewServer(addr string) Server {
 	}
 
 	return Server{
-		router: router,
-		Server: srv,
+		router:      router,
+		Server:      srv,
+		middlewares: []middleware.Middleware{},
+		devMode:     devMode,
 	}
 }
 
 // Handle delegates gorilla mux Handler, and accept a HandlerFactory instead of Handler
 func (s *Server) Handle(path string, hf handler.Factory) *mux.Route {
+	// if devMode is true, read configuration from environment variables
+	var envConfiguration config.TenantConfiguration
+	if s.devMode {
+		envConfiguration = config.TenantConfiguration{}
+		envConfiguration.ReadFromEnv()
+	}
+
 	return s.router.NewRoute().Path(path).Handler(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-		// mock tenant configuration
-		configuration := config.TenantConfiguration{
-			DBConnectionStr: "public",
+		var configuration config.TenantConfiguration
+		if s.devMode {
+			configuration = envConfiguration
+		} else {
+			configuration = config.GetTenantConfig(r)
 		}
 
-		h := hf.NewHandler(configuration)
+		h := hf.NewHandler(r.Context(), configuration)
```

Comment:
If we use r.Context here, we should just remove `handler.Conext` as we discussed before?

It makes the Handler confront to stand interface.